File: SpyKid/guardian/src/main/java/sirs/spykid/guardian/activity/QRActivity.java
package sirs.spykid.guardian.activity;

import androidx.annotation.RequiresApi;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.graphics.Bitmap;
import android.os.Build;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;

import sirs.spykid.guardian.R;
import sirs.spykid.guardian.service.QRGenerator;
import sirs.spykid.util.SharedKey;

@RequiresApi(api = Build.VERSION_CODES.O)
public class QRActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_qr);
        ImageView qrImage = findViewById(R.id.qr_image);
        Intent intent = getIntent();
        SharedKey key = intent.getParcelableExtra("key");
        if(key != null) {
            Bitmap bmp = QRGenerator.qrFromString(key.getEncoded());
            qrImage.setImageBitmap(bmp);
            qrImage.setVisibility(View.VISIBLE);
        }
    }
}

File: SpyKid/guardian/src/main/java/sirs/spykid/guardian/activity/QRActivity.kt
package sirs.spykid.guardian.activity

import android.os.Build
import android.os.Bundle
import android.view.View
import android.widget.ImageView
import androidx.annotation.RequiresApi
import androidx.appcompat.app.AppCompatActivity
import sirs.spykid.guardian.R
import sirs.spykid.guardian.service.QRGenerator
import sirs.spykid.util.SharedKey

@RequiresApi(api = Build.VERSION_CODES.O)
class QRActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_qr)
        val qrImage = findViewById<ImageView>(R.id.qr_image)
        val key = intent.getParcelableExtra<SharedKey>("key")
        if (key != null) {
            val bmp = QRGenerator.qrFromString(key.encoded)
            qrImage.setImageBitmap(bmp)
            qrImage.visibility = View.VISIBLE
        }
    }
}
